normalizer_lib.py

```diff
@@ -1,205 +1,224 @@
-
-SEPARATE_PUNCTUATION = (
-    cdrewrite(
-        pynutil.insert(" "),
-        union("[BOS]", byte.SPACE) + union(NON_GRAPHEME_PUNCT),
-        (union(PUNCTUATION, Verbalizable.verbalizable()) +
-         union("[EOS]", byte.SPACE, Verbalizable.verbalizable())),
-        SIGMA_STAR) @
-    cdrewrite(
-        pynutil.insert(" "),
-        union(Verbalizable.verbalizable(), PUNCTUATION),
-        PUNCTUATION + union("[EOS]", byte.SPACE, PUNCTUATION),
-        SIGMA_STAR))
+        Args:
+            string: A line from a corpus.
 
+        Returns:
+            The line, where invalid tokens have been replaced with <UNK>.
+        """
+        valid_token = (self.initial_punctuation.ques +
+                       self.verbalizable() +
+                       self.final_punctuation.ques)
+        returned: List[str] = []
+        remove_extra_whitespace = (string @ self.remove_extra_whitespace()).optimize().string()
+        split_string = remove_extra_whitespace.split(" ")
+        for token in split_string:
+            if difference(acceptor(token), valid_token).num_states() == 0:
+                returned.append(token)
+            else:
+                returned.append("<UNK>")
+        return " ".join(returned)
+
+
+    def pass_only_valid_sentences(self, string) -> List[str]:
+        """Replaces invalid sentences.
+
+        Args:
+            string: A line from a corpus.
+
+        Returns:
+            The line or the line replaced with <SENTENCE_REJECTED>.
+        """
+        valid_token = (self.initial_punctuation.ques +
+                       self.verbalizable() +
+                       self.final_punctuation.ques)
+        remove_extra_whitespace = (string @ self.remove_extra_whitespace()).optimize().string()
+        split_string = remove_extra_whitespace.split(" ")
+        for token in split_string:
+            if difference(acceptor(token), valid_token).num_states() != 0:
+                return "<SENTENCE_REJECTED>"
+        return string
+
+
+    # Detach punctuation from words
+    # e.g. "Who are you?" -> "Who are you ?"
+
+
+    def detach_punctuation(self) -> Fst:
+        "Detaches punctuation from words."
+        non_grapheme_punct = difference(self.punctuation, self.graphemes)
+        detach_leading_punctuation = cdrewrite(
+            pynutil.insert(" "),
+            union("[BOS]", byte.SPACE) + non_grapheme_punct.plus,
+            union(non_grapheme_punct, self.graphemes, self.numerals),
+            self.SIGMA_STAR)
+        detach_trailing_punctuation = cdrewrite(
+            pynutil.insert(" "),
+            union(non_grapheme_punct, self.graphemes, self.numerals, "/"),
+            self.punctuation.plus + union("[EOS]", byte.SPACE),
+            self.SIGMA_STAR)
+        return (detach_leading_punctuation @ detach_trailing_punctuation).optimize()
+
+
+    # Delete freestanding punctuation
+    # e.g. "hi there ?" -> "hi there
+
+
+    def delete_freestanding_punctuation(self) -> Fst:
```

Comment:
Again doesn't really need to be a method.

Reply:
Agreed! See my comment above about moving all the FSTs into the class init.